### PR TITLE
ShrinkWrapHeight widget

### PR DESCRIPTION
**Pull request by Hixie**
- add debugDescribeSettings to a few classes that were missing it
- fix some minor bugs in RenderShrinkWrapWidth and ShrinkWrapWidth
- introduce RenderShrinkWrapHeight and ShrinkWrapHeight

### CODE REVIEW

File: sky/packages/sky/lib/widgets/basic.dart [hunk 1]
```diff
@@ -277,6 +277,16 @@ class ShrinkWrapWidth extends OneChildRenderObjectWrapper {
...
 
+class ShrinkWrapHeight extends OneChildRenderObjectWrapper {
+  ShrinkWrapHeight({ Key key, Widget child })
+    : super(key: key, child: child);
+
+  RenderShrinkWrapHeight createNode() => new RenderShrinkWrapHeight();
+  RenderShrinkWrapHeight get renderObject => super.renderObject;
+
+  // Nothing to sync, so we don't implement syncRenderObject()
+}
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/basic.dart]: No support for step sizing?

**Comment by abarth** [comment]
Please add a test.  You can use the `layout` test harness.  See `//sky/unit/test/rendering/image_test.dart` for an example.

File: sky/packages/sky/lib/widgets/basic.dart (hunk 1 above)
Comment (Hixie) [inline sky/packages/sky/lib/widgets/basic.dart]: Not unless you have a use case for it...

**Comment by abarth** [comment]
LGTM

**Pull request merged**